mmdet/models/dense_heads/fcos_head.py

```diff
@@ -109,32 +133,44 @@ def _init_layers(self):
 
     def init_weights(self):
         for m in self.cls_convs:
-            normal_init(m.conv, std=0.01)
+            if isinstance(m.conv, nn.Conv2d):
+                normal_init(m.conv, std=0.01)
         for m in self.reg_convs:
-            normal_init(m.conv, std=0.01)
+            if isinstance(m.conv, nn.Conv2d):
+                normal_init(m.conv, std=0.01)
         bias_cls = bias_init_with_prob(0.01)
         normal_init(self.fcos_cls, std=0.01, bias=bias_cls)
         normal_init(self.fcos_reg, std=0.01)
         normal_init(self.fcos_centerness, std=0.01)
 
     def forward(self, feats):
-        return multi_apply(self.forward_single, feats, self.scales)
-
-    def forward_single(self, x, scale):
-        cls_feat = x
-        reg_feat = x
-
-        for cls_layer in self.cls_convs:
-            cls_feat = cls_layer(cls_feat)
-        cls_score = self.fcos_cls(cls_feat)
-        centerness = self.fcos_centerness(cls_feat)
-
-        for reg_layer in self.reg_convs:
-            reg_feat = reg_layer(reg_feat)
-        # scale the bbox_pred of different level
-        # float to avoid overflow when enabling FP16
-        bbox_pred = scale(self.fcos_reg(reg_feat)).float().exp()
-        return cls_score, bbox_pred, centerness
+        cls_scores = []
+        bbox_preds = []
+        centernesses = []
+        for i, x in enumerate(feats):
```

Comment:
Since there is no special operation on a specific fpn layer, is it possible to revert to `forward_single` implementation?

Reply:
Except that there would be another argument (stride) to be passed to `forward_single`.

Reply:
Great idea! Already updated in the latest commit.